integration-tests/test/test_propose.py

```diff
@@ -1,24 +1,58 @@
 import os
 from random import Random
-
-from docker.client import DockerClient
-
-from .common import (
-    CommandLineOptions,
+from .rnode import (
+    extract_validator_stake_from_deploy_cost_str,
+    parse_show_block_output
 )
-from .conftest import testing_context
-from .rnode import docker_network_with_started_bootstrap
-from .wait import (
-    wait_for_approved_block_received_handler_state,
+from .rnode import (
+    Node,
 )
 
 
-def test_propose(command_line_options: CommandLineOptions, random_generator: Random, docker_client: DockerClient) -> None:
-    with testing_context(command_line_options, random_generator, docker_client) as context:
-        with docker_network_with_started_bootstrap(context=context) as bootstrap_node:
-            wait_for_approved_block_received_handler_state(context, bootstrap_node)
-            relative_paths = bootstrap_node.shell_out('sh', '-c', 'ls /opt/docker/examples/*.rho').splitlines()
-            relative_path = random_generator.choice(relative_paths)
-            full_path = os.path.join('/opt/docker/examples', relative_path)
-            bootstrap_node.deploy(full_path)
-            bootstrap_node.propose()
+
+CONTRACT_COST = {
+    "longfast.rho": 98445,
+    "longslow.rho": 3851039,
+    "shortfast.rho": 132,
+    "shortslow.rho": 3785344,
+    "stderr.rho": 126,
+    "stderrAck.rho": 485,
+    "stdout.rho": 126,
+    "stdoutAck.rho": 485,
+    "time.rho": 706,
+    "tut-bytearray-methods.rho": 532,
+    "tut-hash-functions.rho": 1368,
+    "tut-hello-again.rho": 2782,
+    "tut-lists-methods.rho": 648,
+    "tut-maps-methods.rho": 2823,
+    "tut-parens.rho": 399,
+    "tut-philosophers.rho": 6111,
```

Comment:
Is tut-philosophers.rho really deterministic?

Reply:
sorry. Forget to push my update here